messybrainz/db/recording.py

```diff
@@ -148,3 +159,74 @@ def create_recording_clusters():
                 clusters_modified += 1
 
     return clusters_modified, clusters_add_to_redirect
+
+
+def cluster_new_recording(recording_data):
+    """ Tries to associate and cluster newly inserted recording.
+
+    Args:
+        recording_data(dict): Data present in the recording.
+
+    Returns:
+        cluster_id(UUID): Cluster ID for the new recording.
+    """
+
+    try:
+        _send_recording_to_queue(recording_data)
+    except (InternalServerError, ServiceUnavailable) as e:
+        raise
+    except Exception as e:
```

Comment:
this is from LB and is bad code. We should handle exceptions better. I'm not happy with using flask errors in the db module.